src/Aspire.Dashboard/Components/ResourcesGridColumns/GridColumnManager.cs

```diff
@@ -1,54 +1,57 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Text;
 using Aspire.Dashboard.Components.Resize;
 using Aspire.Dashboard.Model;
 
 namespace Aspire.Dashboard.Components;
 
-public class GridColumnManager
+public class GridColumnManager(IEnumerable<GridColumn> columns, DimensionManager dimensionManager)
 {
-    private readonly GridColumn[] _columns;
-    private readonly DimensionManager _dimensionManager;
+    private readonly Dictionary<string, GridColumn> _columnById
+        = columns.ToDictionary(c => c.Name, StringComparers.GridColumn);
 
-    public GridColumnManager(GridColumn[] columns, DimensionManager dimensionManager)
+    /// <summary>
+    /// Gets whether the column is known, visible, and has a width for the current viewport.
+    /// </summary>
+    public bool IsColumnVisible(string columnName)
     {
-        if (columns.DistinctBy(c => c.Name, StringComparers.GridColumn).Count() != columns.Length)
-        {
-            throw new InvalidOperationException("There are duplicate columns");
-        }
-
-        _columns = columns;
-        _dimensionManager = dimensionManager;
+        return _columnById.TryGetValue(columnName, out var column) // Is a known column.
+            && GetColumnWidth(column) is not null                  // Has width for current viewport.
+            && column.IsVisible?.Invoke() is null or true;         // Is visible.
     }
 
-    public bool IsColumnVisible(string columnId)
-    {
-        return GetColumnWidth(_columns.First(column => column.Name == columnId)) is not null;
-    }
-
-    private string? GetColumnWidth(GridColumn column)
+    /// <summary>
+    /// Gets a string that can be used as the value for the grid-template-columns CSS property.
+    /// For example, <c>1fr 2fr 1fr</c>.
+    /// </summary>
+    /// <returns></returns>
+    public string GetGridTemplateColumns()
     {
-        if (column.IsVisible is not null && !column.IsVisible())
-        {
-            return null;
-        }
+        StringBuilder sb = new();
```

Comment:
Repo standard is use var wherever possible.

Reply:
There's a decent mix of both throughout the repo.

- `" = new("` matches 171 files
- `"var \w+ = new "` matches 341 files